src/nwb/NWBFile.cpp

```diff
@@ -30,19 +30,23 @@ NWBFile::NWBFile(const std::string& idText, std::shared_ptr<BaseIO> io)
 
 NWBFile::~NWBFile() {}
 
-void NWBFile::initialize()
+Status NWBFile::initialize()
 {
+  if (isRecording) {
+    return Status::Failure;  // cannot initialize new datasets in recording mode
+  }
+
   if (std::filesystem::exists(io->getFileName())) {
-    io->open(false);
+    return io->open(false);
   } else {
     io->open(true);
-    createFileStructure();
+    return createFileStructure();
   }
 }
 
-void NWBFile::finalize()
+Status NWBFile::finalize()
 {
-  io->close();
+  return io->close();
```

Comment:
Should `finalize` also need to call `stopRecording`?

Reply:
I think that makes sense to call `stopRecording` in `finalize` in case the user has not already called it. Updated this accordingly.